### PR TITLE
Feature/network requests part1

### PR DESCRIPTION
**Pull request by BertoCruz**
# Pull Request
- [x] New feature (adds functionality)
- [x] Refactor

### Summary:
This branch will have the network requests created for the `<App>` and `<movieDetails>` components.
This also refactors our react components structure from functional to class components.
- `<MovieDetails>` -  converted to a class component in order to hold single movie details state + run componentDidMount for network requests upon render.
- `<MovieContainer>` converted to a functional component as this component does not need to hold state

### Please describe the tests that you ran to verify your changes.
- [x] I tested my changes in the browser
- [ ] I tested with Cypress

### Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new bugs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by LaurenBlack5280 — APPROVED** [review]
- [x] Code is clean and maintains SRP
- [x] Great UI
- [x] Network Requests working as expected

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const fetchMovieData = (endpoint) => {
+    return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies${endpoint}`)
+    .then(res => {
+      if(!res.ok) {
+        throw new Error()
```
Comment (hfaerber) [inline src/apiCalls.js:5]: Love that you're checking that the response is ok and throwing an error if not.

File: src/components/App/App.js [hunk 2]
```diff
@@ -3,26 +3,56 @@ import React, { Component } from "react";
...
 
   render() {
+    if (this.state.isLoading) {
+      return <p>Loading...</p>;
+    }
     return (
       <div className="App">
-        <Header randomMovie={this.randomMoviePoster()} />
-        <MovieContainer movieData={this.state.movies} />
+        {/* {this.state.error && <h2>{this.state.error}</h2>} */}
```
Comment (hfaerber) [inline src/components/App/App.js:48]: I see this is commented out now.  Eager to see where/how you end up displaying the error if your app encounters one.

File: src/components/App/App.js [hunk 3]
```diff
@@ -3,26 +3,56 @@ import React, { Component } from "react";
...
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
+      allMovies: movieData.movies,
+      movies: [],
+      movieForHero: {},
+      error: "",
```
Comment (hfaerber) [inline src/components/App/App.js:16]: I like how you're using empty states here (array, object, string).  Its a good way to show what shape the data will take once its there - good dev empathy.

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+
+  window.addEventListener("scroll", changeColor)
+
   return (
-    <header
-      style={{
-        backgroundImage: `url(${randomMovie.backdrop_path})`,
-      }}
-    >
+    <header className={color ? "header header-background" : "header"}>
```
Comment (hfaerber) [inline src/components/Header/Header.js:17]: Consider pulling this ternary logic out of this line for readability.  Can you store it in a variable elsewhere then cleanly reference that variable here?   This is a readability opinion - ultimately YOU get to decide.

File: src/components/MovieCard/MovieCard.js [hunk 5]
```diff
@@ -1,19 +1,24 @@
...
-    )
-}
+const MovieCard = ({ id, title, poster, avgRating, releaseDate, handleClick, submitId }) => {
+  return (
+    <section className="movie-card">
+      <div className="movie-card-poster-container">
+        <span className="avg-rating">🍅 <b>{avgRating}/10</b></span>
+        <img
+          src={poster}
+          alt="movie poster"
```
Comment (hfaerber) [inline src/components/MovieCard/MovieCard.js:11]: Nice work including the alt tag for accessibility - can you use something more dynamic so its a more descriptive alt tag based on each movie?

**Comment by LaurenBlack5280** [comment]
Thank you for the careful review! I turned your suggestions into issues.

**Pull request merged**